sys/include/net/nanocoap.h

```diff
@@ -204,14 +204,14 @@ extern "C" {
 #define COAP_OPT_FINISH_PAYLOAD  (0x0001)
 /** @} */
 
+#define COAP_DATA_PTR(hdr) ((uint8_t*)&(hdr) + 4)
```

Comment:
Use `sizeof(coap_hdr_t)` instead of 4 here.

Reply:
Uh, shame on me for that one...
Will be fixed tomorrow